File: src/main/java/com/kishlaly/ta/model/BlockResult.java
package com.kishlaly.ta.model;

public class BlockResult {

    private QuoteJava lastChartQuote;
    private BlockResultCode code;

    public BlockResult(final QuoteJava lastChartQuote, final BlockResultCode code) {
        this.lastChartQuote = lastChartQuote;
        this.code = code;
    }

    public QuoteJava getLastChartQuote() {
        return this.lastChartQuote;
    }

    public void setLastChartQuote(final QuoteJava lastChartQuote) {
        this.lastChartQuote = lastChartQuote;
    }

    public BlockResultCode getCode() {
        return this.code;
    }

    public boolean isOk() {
        return code == BlockResultCode.OK;
    }

}

File: src/main/kotlin/com/kishlaly/ta/model/BlockResult.kt
package com.kishlaly.ta.model

data class BlockResult(var lastChartQuote: Quote, val code: BlockResultCode) {

    fun isOk() = code == BlockResultCode.OK

    companion object {
        fun configurationError(lastQuote: Quote) = BlockResult(lastQuote, BlockResultCode.CONFIGURATION_ERROR)
    }

}